webhook/credentials.go

```diff
@@ -0,0 +1,197 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	EnvKeyGoogleApplicationCredentials   = "GOOGLE_APPLICATION_CREDENTIALS"
+	EnvValueGoogleApplicationCredentials = "some-path"
+
+	AnnotationGCPServiceAccount      = "giantswarm.io/gcp-service-account"
```

Comment:
I think the only thing worth putting in a flag is maybe the expiration on the token, but I don't really see a need to do it right now. Just a sensible default would do, unless we find a reason we'd want to change it.
The rest of these are just setting up the volume and mount path, which shouldn't really matter, since the way to interact with that file is through the environment variable.